Lib/asyncio/locks.py

```diff
@@ -362,12 +362,12 @@ def _wake_up_next(self):
             waiter = self._waiters.popleft()
             if not waiter.done():
                 waiter.set_result(None)
-                self._wakeup_scheduled = True
+                self._wakeup_scheduled += 1
                 return
 
     def locked(self):
         """Returns True if semaphore can not be acquired immediately."""
-        return self._value == 0
+        return self._value <= 0 or self._wakeup_scheduled
```

Comment:
Let's always return a `bool`, like before:
```suggestion
        return self._value <= 0 or bool(self._wakeup_scheduled)
```